lib/client.c

```diff
@@ -1576,6 +1739,33 @@ SF_STMT *STDCALL snowflake_stmt(SF_CONNECT *sf) {
     return sfstmt;
 }
 
+SF_STMT *STDCALL snowflake_async_stmt(SF_CONNECT *sf, const char *query_id) {
+  if (!sf) {
+    return NULL;
+  }
+
+  SF_STMT *sfstmt = (SF_STMT *)SF_CALLOC(1, sizeof(SF_STMT));
+  if (sfstmt) {
+    _snowflake_stmt_reset(sfstmt);
+    sfstmt->connection = sf;
+    sf_strcpy(sfstmt->sfqid, SF_UUID4_LEN, query_id);
+  }
+
+  get_real_results(sfstmt);
```

Comment:
The main purpose for async query execution is to have non-blocking mode for query execution, so at least we should not call `get_real_results` (which would be blocked until query execution finished) when generating result set from query id. In JDBC `getRealResults` seems being called only when application trying to obtain result metadata or serializable query result.
Personally I would prefer to always return special error code (like `SQL_STILL_EXECUTING` in ODBC) for any API call on a result set (SF_STMT) with async query still running, but I'm OK if you want to keep the same approach as JDBC.